test/test_nn.py

```diff
@@ -13413,6 +13423,50 @@ def test_conv_backend_selection(
         backend_actual = torch._C._select_conv_backend(*inputs)
         self.assertEqual(backend_actual, backend_expected)
 
+        # mkldnn doesn't support float64 for gradcheck.
+        if layout is torch._mkldnn:
+            return
+
+        # Convert to float64 for gradcheck.
+        x = x.to(torch.float64).detach().requires_grad_(True)
+        weight = weight.to(torch.float64).detach().requires_grad_(True)
+        bias = bias.to(torch.float64).detach().requires_grad_(True)
+        inputs = [x, weight, bias, stride, padding, dilation, transposed, output_padding, groups]
+
+        # Autograd function to hook up the general convolution function to convolution_backward
+        # without a derivatives.yaml entry. TODO: Once general forward + backward are hooked up together,
+        # remove this.
+        class MyConv(torch.autograd.Function):
+            @staticmethod
+            def forward(ctx, input, weight, bias, stride, padding, dilation, transposed, output_padding, groups):
+                ctx.save_for_backward(input, weight, bias)
+                ctx.stuff = (stride, padding, dilation, transposed, output_padding, groups)
+                return torch.convolution(input, weight, bias, stride, padding, dilation, transposed,
+                                         output_padding, groups)
+
+            @staticmethod
+            def backward(ctx, grad_output):
+                input, weight, bias = ctx.saved_tensors
+                stride, padding, dilation, transposed, output_padding, groups = ctx.stuff
+                grad_input, grad_weight, grad_bias = torch.ops.aten.convolution_backward(
+                    grad_output, input, weight, bias, stride, padding, dilation, transposed,
+                    output_padding, groups, ctx.needs_input_grad[:3])
+                return grad_input, grad_weight, grad_bias, None, None, None, None, None, None
+
+        convolution = MyConv.apply
+
+        # Set some backend-specfic validation settings.
+        gradcheck_nondet_tol = 0.0
+        if torch.backends.cudnn.is_available():
+            # cuDNN introduces non-determinism
+            gradcheck_nondet_tol = GRADCHECK_NONDET_TOL
+
+        self.assertTrue(gradcheck(convolution, inputs, nondet_tol=gradcheck_nondet_tol))
+
+        # double backward doesn't support bias gradients
+        bias.requires_grad_(False)
+        self.assertTrue(gradgradcheck(convolution, inputs, nondet_tol=gradcheck_nondet_tol))
```

Comment:
I can't see the sample inputs (GitHub won't let me, the file is probably too large). We should make sure the sample inputs are not too large otherwise this test could take a long time to run.

Reply:
Good point. The largest samples are of shape `(2, 6, 7, 8, 9)` with 6048 total elements. Running locally gives me: `Ran 456 tests in 1.711s`